Source/MonoGame.Extended/Sprites/Sprite.cs

```diff
@@ -66,9 +70,9 @@ public TextureRegion2D TextureRegion
                     throw new InvalidOperationException("TextureRegion cannot be null");
 
                 // preserve the origin if the texture size changes
-                var originNormalized = OriginNormalized;
+                //var originNormalized = OriginNormalized;
                 _textureRegion = value;
-                OriginNormalized = originNormalized;
+                OriginNormalized = _textureRegion.Anchor;
```

Comment:
@craftworkgames As I understand, previous lines were for changing origin, if texture size is the same. I've commented these lines and change it a little bit, so it could work for each frame (don't know, if it will affect performance), you can make it optional; in my spritesheet it is necessary to change origin each frame.